### DISCUSSION TITLE
DataGrid commits on first keypress when editing

### DISCUSSION THREAD

**Discussion by npXXX** [AvaloniaUI/Avalonia#14789] [Q&A] [answered]
Hi, 
I'm almost certain I'm overlooking something here. I have a DataGrid with 7 columns, although only 4 of those columns are editable. 3 of the editable columns don't show this behaviour. One of them commits the edit AS SOON as a key is pressed, e.g. I want to edit the content of the column, press the "1" key and it immediately commits the change. The DataTemplate contains a StringFormat, which I don't want while editing, so there's an extra CellEditingTemplate. That column is the only one where I have that, maybe I'm doing something wrong there.

Maybe someone already had a similar problem or has an idea, thank you in advance!

Here's the XAML:
<DataGridTemplateColumn Header="Price" Width="Auto">

                                        <DataGridTemplateColumn.CellTemplate>
                                            <DataTemplate>
                                                <TextBlock Text="{Binding Price, StringFormat='{}{0:0.00 €}'}" HorizontalAlignment="Right" VerticalAlignment="Center"/>
                                            </DataTemplate>
                                        </DataGridTemplateColumn.CellTemplate>
                                        <DataGridTemplateColumn.CellEditingTemplate>
                                            <DataTemplate>
                                                <TextBox Text="{Binding Price}" TextAlignment="Right"/>
                                            </DataTemplate>
                                        </DataGridTemplateColumn.CellEditingTemplate>
                                    </DataGridTemplateColumn>

**Comment by timunie** [comment, marked as answer]
That is how TemplateColumn works unfortunately. In 11.1 you will have an option to set `UpdateSourceTrigger` as needed, but not yet released. It can be consumed from nightly builds however.

**Reply by DarkMortainius** [reply]
I just ran into this issue as well. Any idea when 11.1 is going to be released?

**Reply by timunie** [reply]
We're close to beta1 for 11.1. no more new features go into 11.1
